source/ConditionSet.cpp

```diff
@@ -162,12 +163,12 @@ namespace {
 				value = static_cast<int64_t>(DataNode::Value(str));
 			else
 			{
-				const auto temp = created.find(str);
-				const auto perm = conditions.find(str);
-				if(temp != created.end())
-					value = temp->second;
-				else if(perm != conditions.end())
-					value = perm->second;
+				bool tempHas = created.Has(str);
+				bool permHas = conditions.Has(str);
+				if(tempHas)
+					value = created.Get(str);
+				else if(permHas)
+					value = conditions.Get(str);
```

Comment:
Is this an optimization? Idk how exactly has and find work, but we could be looking for it twice when it's there?

Reply:
You're right, this code is indeed looking for the conditions twice when they are there. I'll have a look if I can improve on this.

Reply:
I added an `HasGet` function which performs the Has and Get functions with a single lookup (and returns a pair with the results).

An alternative approach would be to implement a Find which returns a `ConditionEntry`, but the ConditionEntry will not always exist for derived conditions, and I believe it is performance-wise better not to create it when we are only performing read-access.

Reply:
Looks good to me this way